daemon/daemon.go

```diff
@@ -224,13 +220,18 @@ func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		rsp = rspf(c, r, user)
 	}
 
-	if maintTransmitter, ok := rsp.(maintenanceTransmitter); ok {
+	if rsp, ok := rsp.(*resp); ok {
 		_, rst := st.Restarting()
 		switch rst {
 		case state.RestartSystem:
-			maintTransmitter.transmitMaintenance(errorKindSystemRestart, "system is restarting")
+			rsp.transmitMaintenance(errorKindSystemRestart, "system is restarting")
 		case state.RestartDaemon:
-			maintTransmitter.transmitMaintenance(errorKindDaemonRestart, "daemon is restarting")
+			rsp.transmitMaintenance(errorKindDaemonRestart, "daemon is restarting")
+		}
+		if rsp.Type != ResponseTypeError {
+			st.Lock()
+			rsp.addWarningsToMeta(st.WarningsSummary)
```

Comment:
Curious.. I'd have expected PendingWarnings (WarningsToShow) here.

Reply:
yeah, the discussion on that as reflected on the whiteboard was a bit confusing. Initially we thought of printing the actual warnings after each command (“Warnings show up at the bottom of other commands, but, If more than 3, say WARNING: see `snap  warnings`”), but then as we thought of it more it got cut back to “On commands, if last warning is more recent than last seen show WARNING: There are new messages. See `snap warnings`”. This sending the warning summary supports the latter directly, and the client can implement the former by hitting  the warnings endpoint if the summary indicates it needs to.